src/main/java/com/bookshop/annotation/Isbn.java

```diff
@@ -0,0 +1,25 @@
+package com.bookshop.annotation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import jakarta.validation.ReportAsSingleViolation;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.hibernate.validator.constraints.ConstraintComposition;
+
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = {})
+@ConstraintComposition
+@ReportAsSingleViolation

```

Comment:
I don't understand what this annotations do, but I can guess that some unnecessary (I'm sure @Constraint definitely not needed because you left validatedBy empty).